src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java

```diff
@@ -0,0 +1,78 @@
+/*
+    Copyright 2017 Ericsson AB.
+    For a full list of individual contributors, please see the commit history.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+package com.ericsson.ei.subscriptionhandler;
+
+import java.util.Iterator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.ericsson.ei.jmespath.JmesPathInterface;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+/**
+ * This class represents the mechanism to fetch the rule conditions from the
+ * Subscription Object and match it with the aggregatedObject to check if it is
+ * true.
+ * 
+ * @author xjibbal
+ *
+ */
+
+@Component
+public class RunSubscription {
+
+    @Autowired
+    private JmesPathInterface jmespath;
+
+    static Logger log = (Logger) LoggerFactory.getLogger(RunSubscription.class);
+
+    /**
+     * This method matches every condition specified in the subscription Object
+     * and if all conditions are matched then only the aggregatedObject is
+     * eligible for notification via e-mail or REST POST.
+     * 
+     * @param aggregatedObject
+     * @param requirement
+     * @param subscriptionJson
+     * @return boolean
+     */
```

Comment:
There is a new test case you need to implement since I see a problem with your code. At firs subscription that fails it will stop and not check all the other subscriptions.

When we have an aggregated object and a set of three subscription. First subscription will fail the second will pass and the third will fail. Be sure that all the subscriptions are checked and not only the first one.

The second test is when you have a set of three subscriptions and only the first two pass. All three subscriptions needs to be run. 

Do not loop your self through the subscription set but you need to pass them to the subscriptionHandler so that as much of your code is tested.